docs/modeling/configuring.rst

```diff
@@ -148,7 +132,11 @@ No value generation means that you will always supply a valid value to be saved
 Value generated on add
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Value generated on add means that if you don't specify a value one will be generated for you. This value may be generated client side by EF, or in the database.
+Value generated on add means that if you don't specify a value, one will be generated for you.
+
+If you add an entity to the context that has a value assigned to the primary key property, then EF will attempt to insert that value rather than generating a new one. A property is considered to have a value assigned if it is not assigned the CLR default value (``null`` for ``string``, ``0`` for ``int``, ``Guid.Empty`` for ``Guid``, etc.).
+
+Depending on the database provider being used, values may be generated client side by EF, or in the database. If the value is generated by the database, then EF may assign a temporary value when you add the entity to the context. This temporary value will then be replaced by the database generated value during ``SaveChanges``.
 
```

Comment:
@ajcvickers could you take a read of this section and let me know if you think this is a decent explanation. BTW I think we made good progress on the product here because it is way easier to explain than what we had previously.